stake-pool/program/src/processor.rs

```diff
@@ -845,7 +870,30 @@ impl Processor {
             return Err(ProgramError::IncorrectProgramId);
         }
 
-        let mut stake_pool = State::deserialize(&stake_pool_info.data.borrow())?.stake_pool()?;
+        let mut stake_pool = StakePool::deserialize(&stake_pool_info.data.borrow())?;
+        if !stake_pool.is_initialized() {
+            return Err(StakePoolError::InvalidState.into());
+        }
+
+        // TODO: remove conditional compilation when time travel in tests is possible
+        #[cfg(not(feature = "test-bpf"))]
+        {
+            let stake_acc_state: stake::StakeState =
+                deserialize(&stake_info.data.borrow()).unwrap();
+            let delegation = stake_acc_state.delegation();
+            if let Some(delegation) = delegation {
+                let target_epoch = clock.epoch;
+                let history = None;
```

Comment:
You'll also need to pass in the `StakeHistory`sysvar to calculate this correctly